cmd/juju/model/setcredential.go

```diff
@@ -0,0 +1,188 @@
+type modelCredentialCommand struct {
+	modelcmd.ModelCommandBase
+
+	cloud      string
+	credential string
+
+	newAPIRootFunc            func() (base.APICallCloser, error)
+	newModelCredentialAPIFunc func(base.APICallCloser) ModelCredentialAPI
+	newCloudAPIFunc           func(base.APICallCloser) CloudAPI
+}
+
+func NewModelCredentialCommand() cmd.Command {
+	command := &modelCredentialCommand{
+		newModelCredentialAPIFunc: func(root base.APICallCloser) ModelCredentialAPI {
+			return modelmanager.NewClient(root)
+		},
+		newCloudAPIFunc: func(root base.APICallCloser) CloudAPI {
+			return cloudapi.NewClient(root)
+		},
+	}
+	command.newAPIRootFunc = func() (base.APICallCloser, error) {
+		return command.NewControllerAPIRoot()
+	}
+	return modelcmd.Wrap(command)
+}
+
+// Info implements Command.Info.
+func (c *modelCredentialCommand) Info() *cmd.Info {
+	return &cmd.Info{
+		Name:    "set-credential",
+		Args:    "<cloud name> <credential name>",
+		Purpose: "Change cloud credential for the current or specified model.",
+		Doc:     modelCredentialDoc,
+	}
+}
+
+// SetFlags implements Command.SetFlags.
+func (c *modelCredentialCommand) SetFlags(f *gnuflag.FlagSet) {
+	c.ModelCommandBase.SetFlags(f)
+}
+
+// Init implements Command.Init.
+func (c *modelCredentialCommand) Init(args []string) error {
+	if len(args) != 2 {
+		return errors.Errorf("Usage: juju set-credential [options] <cloud name> <credential name>")
+	}
+	if !names.IsValidCloud(args[0]) {
+		return errors.NotValidf("cloud name %q", args[0])
+	}
+	if !names.IsValidCloudCredentialName(args[1]) {
+		return errors.NotValidf("cloud credential name %q", args[1])
+	}
+	c.cloud = args[0]
+	c.credential = args[1]
+	return nil
+}
+
+// Run implements Command.Run.
+func (c *modelCredentialCommand) Run(ctx *cmd.Context) error {
+	fail := func(e error) error {
+		ctx.Infof("Failed to change model credential: %v", e)
+		return e
+	}
+
+	root, err := c.newAPIRootFunc()
+	if err != nil {
+		return fail(errors.Annotate(err, "opening API connection"))
+	}
+	defer root.Close()
+
+	accountDetails, err := c.CurrentAccountDetails()
+	if err != nil {
+		return fail(errors.Annotate(err, "getting current account"))
+	}
+	userTag := names.NewUserTag(accountDetails.User)
+	cloudTag := names.NewCloudTag(c.cloud)
+	credentialTag, err := common.ResolveCloudCredentialTag(userTag, cloudTag, c.credential)
+	if err != nil {
+		return fail(errors.Annotate(err, "resolving credential tag"))
+	}
+
+	cloudClient := c.newCloudAPIFunc(root)
+	defer cloudClient.Close()
+
+	remote := false
+	remoteCredentials, err := cloudClient.UserCredentials(userTag, cloudTag)
+	if err != nil {
+		// This is ok - we can proceed with local ones anyway.
+		ctx.Infof("Could not determine if there are remote credentials for the user: %v", err)
+	} else {
+		for _, credTag := range remoteCredentials {
+			if credTag == credentialTag {
+				remote = true
+				ctx.Infof("Found credential remotely, on the controller. Not looking locally...")
```

Comment:
Just "Found credential on the controller." might be clearer?

Reply:
There is a debate about whether people know that there is a controller and what it is. Saying "remotely" covers the situation where saying 'controller' may raise more questions than provide answers...